### PR TITLE
Improvements and Additions to Alert Testing Stack

### PR DESCRIPTION
**Pull request by dgoodwin**
[TRT-1235](https://issues.redhat.com//browse/TRT-1235) [TRT-934](https://issues.redhat.com//browse/TRT-934)

This PR continues work to resume some form of automated testing on alerts in a couple specific scenarios. 

## Removed Duplicated Alert Testing stack

Two tests are being removed in favor of a newer invariant based test we added some time ago which works off intervals we generated by querying prometheus, rather than repeated live prometheus queries to get the same data. This new style of testing allows us to run the alert tests against a given intervals file, which proved very useful on this PRs development.

The tests being removed:

`[sig-instrumentation][Late] Alerts shouldn't report any unexpected alerts in firing or pending state [apigroup:config.openshift.io] [Suite:openshift/conformance/parallel]`
- Ginko test run Late during e2e, live queries prometheus for the duration of
  the run, skips any alerts that have their own test, and flakes if any
  are found. It cannot fail.
- Flake rate is about 80%.

`[sig-arch] Check if alerts are firing during or after upgrade success`
- Monitor test run after upgrade.
- Uses basically the same approach/code as above.
- Flake rate is 97%.

Removing these two tests dramatically simplifies our stack so we can begin
working on resurrecting some form of alert testing in just one spot.

The backstop test which replaces them is: `[sig-trt][invariant] No alerts without an explicit test should be firing/pending more than historically`

NOTE: A similarly named test that can fail and does catch real problems
is NOT being removed, this one is looking for alerts at the specific point in time it runs, i.e. nothing should be firing if we're done our e2e run:

`[sig-instrumentation] Prometheus [apigroup:image.openshift.io] when installed on the cluster shouldn't report any alerts in firing state apart from Watchdog and AlertmanagerReceiversNotConfigured [Early][apigroup:config.openshift.io] [Skipped:Disconnected] [Suite:openshift/conformance/parallel]`

## New Test To Ensure New Alerts Don't Fire

Added a test `[sig-trt][invariant] No new alerts should be firing` which checks for any occurrence of a firing non-info alert we don't know about (not in our data file), or is in our data file but is less than two weeks old.  This was a feature of the planned design that was intended to give us a window to catch someone adding a new alert, that is firing in CI already. (we don't want things to get worse)

This test is set to flake, I'll move to a fail once I can see if run for a bit and confirm this is safe.

## Ignore TelemeterClientFailures Alerts

This indicates a problem with an external service, it does not affect other tests, it's pending all the time and makes it to firing plenty often. Not an alert we're concerned with in testing, after speaking with monitoring team this is being removed. (Trevor is actually driving us towards disabling telemeter in CI clusters)

## Extract More Alert Data into Structured Intervals

Expanded our structured alert intervals to now parse our the alert severity and level.

## Read More Data from Alerts Query Results

The alerts query_results.json will now include more data such as first/last observed, start reading this info out and using minimally.

## Fix The 100% Flake Rate on Existing Backstop Test

The test we're keeping called out above was a 100% flake, I believe because it did not consider the alerts we know fire on every CI cluster, AlertManagerNotConfigured and Watchdog. This alert now shares the list of alerts we do not care about with the remaining prometheus ginko test.

## Next Steps

We have no plan to bring back historical alert data testing as we believe we cannot possibly keep up with the noise it generates. Right now the focus is on making sure nobody adds new alerts that are already firing in our fleet, and hopefully never seeing that happen for SD, as well as hunting down other pinned alerts we can work on for SD.

### CODE REVIEW

File: pkg/monitortestlibrary/allowedalerts/basic_alert.go [hunk 1]
```diff
@@ -111,6 +112,11 @@ func (a *alertBuilder) neverFail() *alertBuilder {
 	return a
 }
 
+func (a *alertBuilder) alwaysFlake() *alertBuilder {
+	a.allowanceCalculator = alwaysFlake()
+	return a
+}
+
```
Comment (dgoodwin) [inline pkg/monitortestlibrary/allowedalerts/basic_alert.go:119]: The use of this was removed, but I left the alwaysFlake functionality in so it's visible when we start working on more alerts to help SD out.

**Comment by stbenjam** [comment]
Do you consider the removed tests to be comparable to `[sig-trt][invariant] No alerts without an explicit test should be firing/pending more than historically`? If so we may want to track it as a [rename](https://github.com/openshift-eng/ci-test-mapping/pull/38).  If it's substantially different we can just leave it.

**Comment by stbenjam** [comment]
Otherwise code looks good to me, test seems to be passing in the jobs

/lgtm

**Comment by stbenjam** [comment]
Also `[sig-trt][invariant] No alerts without an explicit test should be firing/pending more than historically` fires A LOT, more than [99% of the time](https://sippy.dptools.openshift.org/sippy-ng/tests/4.15?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522never-stable%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522aggregated%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522%255Bsig-trt%255D%255Binvariant%255D%2520No%2520alerts%2520without%2520an%2520explicit%2520test%2520should%2520be%2520firing%252Fpending%2520more%2520than%2520historically%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&sort=desc&sortField=current_flake_percentage&view=Flakes).  It seems like we need to fix up that test too to allow exceptions, like AlertmanagerReceiversNotConfigured

**Pull request merged**